vitrivr-engine-query/src/main/kotlin/org/vitrivr/engine/query/parsing/QueryParser.kt

```diff
@@ -66,7 +67,14 @@ class QueryParser(val schema: Schema) {
         /* Extract necessary information. */
         val operation = description.operations[operatorName] as? RetrieverDescription ?: throw IllegalArgumentException("Operation '$operatorName' not found in information need description.")
         val input = description.inputs[operation.input] ?: throw IllegalArgumentException("Input '${operation.input}' for operation '$operatorName' not found")
-        val field = operation.field?.let { this.schema[it] }
+        val fieldAndAttributeName: Pair<String,String?> = if(operation.field.contains(".")){
+            val f = operation.field.split(".").getOrNull(0) ?: throw IllegalArgumentException("Field name in dot notation FIELD.ATTRIBUTE requires both, field and attribute")
+            val a = operation.field.split(".").getOrNull(1) ?: throw IllegalArgumentException("Field name in dot notation FIELD.ATTRIBUTE requires both, field and attribute")
```

Comment:
I'm still not convinced that this is correct. If you have a struct field name containing a `.` - which you probably shouldn't but who knows - it will cut off anything after the second `.`.  I'd probably just use `substringBefore(".")` for `f` and `substringAfter(".")` for `a`.

Reply:
Well, I addressed this now in two ways: First, I follow your suggestion with the `substring(Before|After)` and I prevent from creating fields that have a dot in their name in the future.
This might anyways be unfavourable, depending on the storage layer and therefore I think this is a limitation to be accepted.